### DISCUSSION TITLE
Create Excel file in Memory and add it as attachment

### DISCUSSION THREAD

**Discussion by reinoutD** [dfinke/ImportExcel#1304] [Q&A]
I am trying to create the xlsx file to memory:
`$xlsx = $SecurityGroups |Export-Excel -KillExcel -ClearSheet  -Show  -AutoSize  -PivotRows GroupName,GroupMember  -includePivotTable -PivotData GroupMember -WorksheetName "Groepen en Leden"  -HideSheet "Groepen en Leden" -PivotTableName "GroepenEnLeden" -PassThru`

After the excel document object is created i convert it to a Base64 string:
`$bytes = [System.Text.Encoding]::UTF8.GetBytes($xlsx)`
`$b64 = [System.Convert]::ToBase64String($bytes)`

However, when i receive the mail it doesn't get me an Excel file, but a text file with this content: OfficeOpenXml.ExcelPackage

Is there a way to generate the excel file in memory, so i can attach it from the memory as attachment in the mail via Powershell?

The reason is that i do have a Powershell runspace, but i do not have the possibility to save it as a file. I work in an Azure Runbook.

**Comment by ninmonkey** [comment]
```ps1
$bytes = [System.Text.Encoding]::UTF8.GetBytes($xlsx)
```
This line is the problem. `$xlsx` isn't bytes. It's coercing the type name, which matches your email
```
Pwsh> $xlsx.GetType().ToString()
OfficeOpenXml.ExcelPackage
```
Maybe try (there's a few functions using Get-Member)
```ps1
$xlsx.GetAsByteArray()
```

**Reply by dfinke** [reply]
Yes, the C# only supports reading from the filesystem.
